src/main/java/org/prebid/server/bidder/admixer/AdmixerBidder.java

```diff
@@ -85,23 +85,31 @@ private ExtImpAdmixer parseImpExt(Imp imp) {
         } catch (IllegalArgumentException e) {
             throw new PreBidException(String.format("Wrong Admixer bidder ext in imp with id : %s", imp.getId()));
         }
-        if (StringUtils.length(extImpAdmixer.getZone()) != 36) {
+        if (StringUtils.length(extImpAdmixer.getZone()) < 32 || StringUtils.length(extImpAdmixer.getZone()) > 36) {
             throw new PreBidException("ZoneId must be UUID/GUID");
         }
 
         return extImpAdmixer;
     }
 
     private Imp processImp(Imp imp, ExtImpAdmixer extImpAdmixer) {
-        final Double extImpFloor = extImpAdmixer.getCustomFloor();
-        final BigDecimal customFloor = extImpFloor != null ? BigDecimal.valueOf(extImpFloor) : BigDecimal.ZERO;
+        final BigDecimal customFloor = extImpAdmixer.getCustomFloor();
+        final BigDecimal impFloor = isValidBidFloor(imp.getBidfloor()) ? imp.getBidfloor() : null;
+        final BigDecimal bidfloor = isValidBidFloor(customFloor) && !isValidBidFloor(impFloor)
+                ? customFloor
+                : impFloor;
```

Comment:
1. Extract to separate method.
2. Why you're checking `imp.getBidfloor()` two times?